artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java

```diff
@@ -162,6 +164,14 @@ public int getReconnectID() {
       return sessionChannel.getReconnectID();
    }
 
+   public boolean isKilled() {
+      return killed;
+   }
+
+   public void kill(boolean killed) {
```

Comment:
Sorry, I wasn't clear.  When I mentioned `kill()` I was thinking:

```
public void kill() {
   this.killed = true;
}
```

This way the caller isn't directly modifying the variable.  But no big deal.